cpp/test/sg/rf_treelite_test.cu

```diff
@@ -0,0 +1,397 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuda_utils.h>
+#include <gtest/gtest.h>
+#include <sys/stat.h>
+#include <test_utils.h>
+#include <treelite/c_api.h>
+#include <treelite/c_api_runtime.h>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <limits>
+#include <string>
+#include "decisiontree/decisiontree_impl.h"
+#include "ml_utils.h"
+#include "randomforest/randomforest.hpp"
+
+namespace ML {
+
+using namespace MLCommon;
+
+template <typename T>  // template useless for now.
+struct RfInputs {
+  int n_rows;
+  int n_cols;
+  int n_trees;
+  float max_features;
+  float rows_sample;
+  int n_inference_rows;
+  int max_depth;
+  int max_leaves;
+  bool bootstrap;
+  bool bootstrap_features;
+  int n_bins;
+  int split_algo;
+  int min_rows_per_node;
+  CRITERION split_criterion;
+};
+
+template <typename T>
+::std::ostream& operator<<(::std::ostream& os, const RfInputs<T>& dims) {
+  return os;
+}
+
+template <typename T, typename L>
+class RfTreeliteTestCommon : public ::testing::TestWithParam<RfInputs<T>> {
+ protected:
+  void convertToTreelite() {
+    // Test the implementation for converting fitted forest into treelite format.
+    ModelHandle model;
+    build_treelite_forest(&model, forest, params.n_cols, task_category);
+
+    std::string test_name =
+      ::testing::UnitTest::GetInstance()->current_test_info()->name();
+    // Get the test index from Google current_test_info.
+    // The test index is the string after '/' in test_name.
+    std::string index_str =
+      test_name.substr(test_name.find("/") + 1, test_name.length());
+
+    // Create a directory if the test is the first one in the test case.
+    // https://stackoverflow.com/questions/12510874/how-can-i-check-if-a-directory-exists.
+    std::string create_dir = "mkdir " + test_dir;
+    struct stat info;
+    if (!(stat(test_dir.c_str(), &info) == 0 && S_ISDIR(info.st_mode))) {
+      ASSERT(system(create_dir.c_str()) == 0, "Call %s fails.",
```

Comment:
Please use the `mkdir()` system call (rather than a utility with the same name) to create a directory.

If you want, you can introduce the `SYSCALL_CHECK()` macro to ensure that system calls do not generate an error, but this is entirely up to you.